### ISSUE TITLE
Add a "Share" Button

### ISSUE THREAD

**Issue by pasekit** [BookStackApp/BookStack#4932]
### Describe the feature you'd like

The problem with using Bookstack as a PWA is that you cannot simply copy the link to the page. On a desktop, it is still possible to get the link via the browser menu, but this actually happens outside the app. However, this is not possible on a smartphone.

My current workaround is to go out of the page and then press and hold the link to the page.

My suggestion:
Add a "Share" button below the export button. Perhaps with several options as with exporting, such as copy link, share in another app.

### Describe the benefits this would bring to existing BookStack users

You always work completely in the web app and do not have to switch to browser functions. In my humble opinion, sharing pages is also a common use.

### Can the goal of this request already be achieved via other means?

Yes, for desktop clients via the three-point menu> More tools> Use copy link

Alternatively, navigate out of the page and then copy the link from it.

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

### How long have you been using BookStack?

3 months to 1 year

**Comment by Man-in-Black** [comment]
Dan just helped me with the same question over on Discord and got a perfect solution for me, because I got the same problem when I want to copy the URL.
The discussion can be found here: https://discord.com/channels/578552496637739008/1172518752751988816

For the feature to work you need to use the visual theme system.
Create a theme and within that theme create the following file and folder: `entities.share-link.blade.php`
Put in the following code:
```html
<button type="button"
        id="share-link-button"
        data-success-text="Link copied to clipboard!"
        class="icon-list-item text-link">
    <span>@icon('share')</span>
    <span>{{ trans('common.share') }}</span>
</button>
<script nonce="{{ $cspNonce }}">
    (async function() {
        const shareButton = document.getElementById('share-link-button');
        shareButton.addEventListener('click', event => {
           copyTextToClipboard(window.location.href);
           window.$events.success(shareButton.dataset.successText);
        });

        async function copyTextToClipboard(text) {
            if (window.isSecureContext && navigator.clipboard) {
                await navigator.clipboard.writeText(text);
                return;
            }

            // Backup option where we can't use the navigator.clipboard API
            const tempInput = document.createElement('textarea');
            tempInput.style = 'position: absolute; left: -1000px; top: -1000px;';
            tempInput.value = text;
            document.body.appendChild(tempInput);
            tempInput.select();
            document.execCommand('copy');
            document.body.removeChild(tempInput);
        }
    })()
</script>
```

Now you need to add a line to each `show.blade.php` of the pages, books, chapters and shelves. I've put this directly under the export button:
```php
            @if($watchOptions->canWatch() && !$watchOptions->isWatching())
                @include('entities.watch-action', ['entity' => $page])
            @endif
            @if(user()->hasAppAccess())
                @include('entities.favourite-action', ['entity' => $page])
            @endif
            @if(userCan('content-export'))
                @include('entities.export-menu', ['entity' => $page])
            @endif
            @include('entities.share-link', ['entity' => $page])
        </div>

    </div>
@stop
```
look for the link with the "share-link" in it.

Thats all, now you have a "share" button in your instance:
![image](https://github.com/BookStackApp/BookStack/assets/9048534/4841a814-8567-444f-886b-834deb9a74b3)

**Comment by Tuphal** [comment]
this is exactly what I'm looking for.
a "Share"-Button in general would be really nice.

**Comment by mikehak74** [comment]
For me it would also be nice to have this in the Default template.

**Comment by Man-in-Black** [comment]
I've done exactly this in the past with help of Dan:
https://bookstack.jelinek-rz.de/books/bookstack/page/url-mit-mausklick-kopieren

**Comment by bendem** [comment]
Isn't that implemented directly in all mobile browsers? Why duplicate that feature in every website when your browser has that easily accessible and integrated with other apps?

![Image](https://github.com/user-attachments/assets/df897dcf-d32f-4d56-b983-acd92491ff1a)

![Image](https://github.com/user-attachments/assets/b79fcd96-5b02-4965-b133-3fe33bce4275)

**Comment by pasekit** [comment]
no, if you use bookstack as a pwa app there is no browser toolbar visible.

> Isn't that implemented directly in all mobile browsers? Why duplicate that feature in every website when your browser has that easily accessible and integrated with other apps?
> 
> ![Image](https://github.com/user-attachments/assets/df897dcf-d32f-4d56-b983-acd92491ff1a)
> 
> ![Image](https://github.com/user-attachments/assets/b79fcd96-5b02-4965-b133-3fe33bce4275)

**Comment by bendem** [comment]
I've never found the need to install apps as PWA, I just tried and you indeed do not have the toolbar to share, a quick workaround is to use the self link in the breadcrumb :

![Image](https://github.com/user-attachments/assets/31377611-bf03-41a8-a5d3-b818fa25b781)

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/4932
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
